Automattic/jetpack
# Color schemes: Decouple sidebar notice styles

Required by https://github.com/Automattic/jetpack/pull/36797

## Proposed changes:

We [re-enqueue the default styles for the Core color schemes on untangled sites](https://github.com/Automattic/jetpack/blob/73b9847b56e3c7346f88be0bbab928a29eadfb4d/projects/packages/jetpack-mu-wpcom/src/features/admin-color-schemes/admin-color-schemes.php#L143-L150) to make sure they take precedence over our custom overrides which are only intended for "tangled" sites.

However, now that we are adding sidebar notices to untangled sites as well (https://github.com/Automattic/jetpack/pull/36797), we need to keep some of these overrides to correctly display them across all the different color schemes. 

To do that, we're decoupling the specific styles for sidebar notices in a separate file, so we can load them independently when needed without loading the rest of the overrides.

Before | After
--- | ---
<img width="399" alt="Screenshot 2024-04-23 at 12 44 54" src="https://github.com/Automattic/jetpack/assets/1233880/5e3ad7db-3d84-48c1-8750-978a77141646"> | <img width="487" alt="Screenshot 2024-04-23 at 12 48 13" src="https://github.com/Automattic/jetpack/assets/1233880/c70092a0-cdfb-4c01-b37c-14875f60a121">


Please note that, although this PR might seem huge at first sight, very few things have changed, so have this in mind when reviewing the changes:
- All SCSS variables have been moved from `colors.scss` to `_variables.scss`
- Created a new `sidebar-notice.scss` file to maintain there the specific styles for sidebar notices
- Refactored `colors.scss` to use the new `_variables.scss` and `sidebar-notice.scss` files

### Other information:


## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:

- Set up a WP.com test site with the default admin interface:
  - Simple sites
    - Apply these changes to your sandbox using the instructions from the auto-generated comment below.
    - Sandbox a simple site
    - Open `wp shell` and run this command: `update_blog_option( blog_id, 'wpcom_admin_interface', 'default' );`
  - Atomic sites
    - Apply these changes to your Atomic site using the instructions from the auto-generated comment below.
    - Go to wordpress.com/hosting-config
    - Select your site
    - Change the admin interface style to Default
- Go to wordpress.com/me/account
- Select the Sunrise color scheme and save your settings
- Go to `/wp-admin`
- Make sure the sidebar notice has the expected colors:
<img width="556" alt="Screenshot 2024-04-23 at 12 24 00" src="https://github.com/Automattic/jetpack/assets/1233880/a09a8258-e984-44f0-8e06-25294b3f8d2d">

- Activate the classic admin interface:
  - Simple sites
     - Open `wp shell` and run these commands: `update_blog_option( blog_id, 'wpcom_admin_interface', 'wp-admin' );
update_blog_option( blog_id, 'wpcom_classic_early_release', 1 );`
  - Atomic sites
    - Go to wordpress.com/hosting-config
    - Select your site
    - Change the admin interface style to Classic
    - Go to <SITE_DOMAIN>/_cli and run this command: `wp option update wpcom_classic_early_release 1`
- Go to `/wp-admin/profile.php`
- Select the Sunrise color scheme and save your settings
- Upon reloading, make sure the admin top bar does not have any unexpected override – it should match the default look and feel of Core (e.g. there shouldn't be a bottom border):

Core | Expected | Unexpected
--- | --- | ---
<img width="589" alt="Screenshot 2024-04-23 at 12 27 40" src="https://github.com/Automattic/jetpack/assets/1233880/57292744-bd2e-4124-b32d-dd224b9ed938"> | <img width="713" alt="Screenshot 2024-04-23 at 12 27 52" src="https://github.com/Automattic/jetpack/assets/1233880/1af26a79-048a-4f7d-9419-2542fa8e44db"> | <img width="731" alt="Screenshot 2024-04-23 at 12 30 47" src="https://github.com/Automattic/jetpack/assets/1233880/ec126b73-657d-45a5-99da-4670d75fece8">





- Follow the testing instructions from https://github.com/Automattic/jetpack/pull/36797 and make sure that when the sidebar notice styles are loaded, the admin top bar still does not have any unexpected override
